Framework/Algorithms/test/LineProfileTest.h

```diff
@@ -390,6 +391,153 @@ class LineProfileTest : public CxxTest::TestSuite {
+    std::vector<double> verticalBinEdges(nHist + 1);
+    for (size_t i = 0; i < verticalBinEdges.size(); ++i) {
+      verticalBinEdges[i] = static_cast<double>(i);
+    }
+    auto vertAxis = make_unique<BinEdgeAxis>(verticalBinEdges);
+    inputWS->replaceAxis(1, vertAxis.release());
+    LineProfile alg;
+    // Don't put output in ADS by default
+    alg.setChild(true);
+    alg.setRethrows(true);
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", inputWS))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setPropertyValue("OutputWorkspace", "_unused_for_child"))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("Direction", "Vertical"))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("Centre", static_cast<double>(nBins) / 2))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("HalfWidth", 3.0))
+    TS_ASSERT_THROWS_NOTHING(alg.execute())
+    TS_ASSERT(alg.isExecuted())
+
+    MatrixWorkspace_sptr outputWS = alg.getProperty("OutputWorkspace");
+    TS_ASSERT(outputWS);
+    TS_ASSERT(outputWS->isHistogramData())
+    TS_ASSERT(!outputWS->isDistribution())
+  }
+
+  void
+  test_vertical_point_data_input_gives_nondistribution_point_data_output() {
+    const size_t nHist = 13;
+    const size_t nBins = 23;
+    MatrixWorkspace_sptr inputWS = create2DWorkspace154(nHist, nBins);
+    const auto vertAxis = inputWS->getAxis(1);
+    TS_ASSERT_EQUALS(vertAxis->length(), inputWS->getNumberHistograms())
+    LineProfile alg;
+    // Don't put output in ADS by default
+    alg.setChild(true);
+    alg.setRethrows(true);
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", inputWS))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setPropertyValue("OutputWorkspace", "_unused_for_child"))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("Direction", "Vertical"))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("Centre", static_cast<double>(nBins) / 2))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("HalfWidth", 3.0))
+    TS_ASSERT_THROWS_NOTHING(alg.execute())
+    TS_ASSERT(alg.isExecuted())
+
+    MatrixWorkspace_sptr outputWS = alg.getProperty("OutputWorkspace");
+    TS_ASSERT(outputWS);
+    TS_ASSERT(!outputWS->isHistogramData())
+    TS_ASSERT(!outputWS->isDistribution())
+  }
+
+  void test_vertical_profile_from_distribution_multiplies_by_bin_widths() {
+    const size_t nHist{4};
+    const size_t nBins{3};
+    const BinEdges edges{{0., 0.1, 1.1, 11.1}};
+    const Frequencies frequencies{3., 2., 1.};
+    const Histogram histogram(edges, frequencies);
+    MatrixWorkspace_sptr inputWS = create<Workspace2D>(nHist, histogram);
+    LineProfile alg;
+    // Don't put output in ADS by default
+    alg.setChild(true);
+    alg.setRethrows(true);
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
```

Comment:
I like your testing, one question: test_Init() already checks generally if the algorithm is initialized after initialization. You decided to (always?) check this again. Is this needed?
Maybe test_Init() can check initialization for child algorithms as well, if needed?